cider/exec/nextgen/jitlib/llvmjit/LLVMJITFunction.cpp

```diff
@@ -161,6 +164,67 @@ JITValuePointer LLVMJITFunction::emitJITFunctionCall(
   }
 }
 
+// look up a runtime function based on the name
+JITValuePointer LLVMJITFunction::emitRuntimeFunctionCall(
+    const std::string& fname,
+    const JITFunctionEmitDescriptor& descriptor) {
+  auto func = module_.module_->getFunction(fname);
+  if (!func) {
+    LOG(FATAL) << "Function: " << fname << " does not exist.";
+  }
+  cloneFunctionRecursive(func);
+
+  llvm::SmallVector<llvm::Value*, JITFunctionEmitDescriptor::DefaultParamsNum> args;
+  args.reserve(descriptor.params_vector.size());
+  for (auto jit_value : descriptor.params_vector) {
+    LLVMJITValue* llvmjit_value = static_cast<LLVMJITValue*>(jit_value);
+    args.push_back(llvmjit_value->llvm_value_);
+  }
+
+  llvm::Value* ans = ir_builder_->CreateCall(func, args);
+  return std::make_unique<LLVMJITValue>(
+      descriptor.ret_type, *this, ans, "ret", JITBackendTag::LLVMJIT, false);
+}
+
+void LLVMJITFunction::cloneFunctionRecursive(llvm::Function* fn) {
+  CHECK(fn);
+  if (!fn->isDeclaration()) {
+    return;
+  }
+  // Get the implementation from the runtime module.
+  auto func_impl = module_.external_module_->getFunction(fn->getName());
+  CHECK(func_impl) << fn->getName().str();
+  if (func_impl->isDeclaration()) {
+    return;
+  }
+
+  auto DestI = fn->arg_begin();
```

Comment:
Why is this variable named liked this? should be `dest_i` to keep format consistent?

Reply:
i ll change it to a easy understanding name